### PR TITLE
TS Strict Aria utils

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 
Don't review until we get https://github.com/adobe/react-spectrum/pull/5569 in. This will complete that work.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-aria/utils/src/getScrollParent.ts [hunk 1]
```diff
@@ -11,15 +11,16 @@
  */
 
 export function getScrollParent(node: Element, checkForOverflow?: boolean): Element {
-  if (isScrollable(node, checkForOverflow)) {
-    node = node.parentElement;
+  let scrollableNode: Element | null = node;
```
Comment (ktabors) [inline packages/@react-aria/utils/src/getScrollParent.ts:14]: Is this because we can't change the API by making it `getScrollParent(node: Element | null,` instead?

File: packages/@react-aria/utils/src/useSyncRef.ts [hunk 2]
```diff
@@ -23,7 +23,9 @@ export function useSyncRef<T>(context?: ContextValue<T> | null, ref?: RefObject<
     if (context && context.ref && ref) {
       context.ref.current = ref.current;
       return () => {
-        context.ref.current = null;
+        if (context.ref) {
+          context.ref.current = null;
```
Comment (ktabors) [inline packages/@react-aria/utils/src/useSyncRef.ts:27]: What's the reason for this making this an `if` check instead of something like `context?.ref?.current = null;`?

File: packages/@react-aria/utils/src/useViewportSize.ts [hunk 3]
```diff
@@ -57,7 +57,7 @@ export function useViewportSize(): ViewportSize {
 
 function getViewportSize(): ViewportSize {
   return {
-    width: visualViewport?.width || window.innerWidth,
-    height: visualViewport?.height || window.innerHeight
+    width: (visualViewport && visualViewport.width) || window.innerWidth,
+    height: (visualViewport && visualViewport?.height) || window.innerHeight
```
Comment (ktabors) [inline packages/@react-aria/utils/src/useViewportSize.ts:61]: Why did you add the `visualViewport &&` on both, but only remove the `?` on one of these?

File: packages/@react-aria/utils/src/useSyncRef.ts (hunk 2 above)
Comment (snowystinger) [reply packages/@react-aria/utils/src/useSyncRef.ts:27]: you can't assign to an optional chain

File: packages/@react-aria/utils/src/getScrollParent.ts (hunk 1 above)
Comment (snowystinger) [reply packages/@react-aria/utils/src/getScrollParent.ts:14]: node should be defined at this point in time in the API, but node.parentElement might be null, so that's where the type comes from

File: packages/@react-aria/utils/src/useViewportSize.ts [hunk 4]
```diff
@@ -57,7 +57,7 @@ export function useViewportSize(): ViewportSize {
 
 function getViewportSize(): ViewportSize {
   return {
-    width: visualViewport?.width || window.innerWidth,
-    height: visualViewport?.height || window.innerHeight
+    width: (visualViewport && visualViewport.width) || window.innerWidth,
```
Comment (snowystinger) [inline packages/@react-aria/utils/src/useViewportSize.ts:60]:
```suggestion
    width: (visualViewport && visualViewport?.width) || window.innerWidth,
```

**Review by ktabors — APPROVED** [review]

**Review by ktabors — APPROVED** [review]

**Review by LFDanLu — APPROVED** [review]
